src/libs/api/middleware.http-error-handler.ts

```diff
@@ -0,0 +1,50 @@
+import { Event } from './types';
+import { jsonSafeParse, normalizeHttpResponse } from '@middy/util';
+import middy from '@middy/core';
+
+// enhance the middy HTTP error handler by incorporating a condition to forward protocolink errors.
+// https://github.com/middyjs/middy/tree/main/packages/http-error-handler
+export default function cacheMiddleware(): middy.MiddlewareObj<Event, any, any> {
+  return {
+    onError: async (request) => {
+      if (request.response !== undefined) return;
+      console.error(request.error);
+
+      // forwarding the protocolink API Axios error
+      if (request.error.response && request.error.expose === undefined) {
```

Comment:
Do we check only `undefined` for `request.error.expose`? Is it better to use `!request.error.expose`?

Reply:
Similar to the previous explanation, handle the situation when `expose` is not a boolean value. This is because it's possible that somewhere else in the code, expose has already been set to false, and in such cases, we should expect it to trigger an HTTP 500 error.